mono/sgen/sgen-alloc.c

```diff
@@ -290,6 +299,10 @@ sgen_alloc_obj_nolock (GCVTable vtable, size_t size)
 		mono_atomic_store_seq (p, vtable);
 	}
 
+done:
+	if (G_UNLIKELY (dump_on_oom_enabled () && !p))
```

Comment:
There's a `if (G_LIKELY (p))` before this condition. Couldn't we use it instead?